File: apps/web/src/core/kurslabel.js
// Wie ein Kurs beschriftet wird — an EINER Stelle.
//
// Erst das Fach, dann der Kursname: „Mathe · 7.5". Im Kalender steht immer die
// Frage „was habe ich jetzt?" vor „mit wem?", und der Kursname allein
// beantwortet sie nicht — er ist frei gewaehlt („7.5", „Gruppe rot", „M7b")
// und sagt bei der Haelfte der Konten gar kein Fach.
//
// Warum ein eigenes Modul: dieselbe Beschriftung entsteht in der Oberflaeche
// (Kalender, Stundenplan) und im ICS-Feed, den Apple und Outlook anzeigen.
// Zwei Fassungen davon hiessen: derselbe Termin heisst im Browser anders als
// im Handykalender. Die Server-Seite hat ihr Gegenstueck in
// `_kurs_label` (apps/api/app/routers/kalender.py) und muss mitgezogen werden.

/** "Mathe · 7.5" — Fach zuerst, Kursname dahinter. */
export function kursLabel(kurs) {
  if (!kurs) return "";
  const fach = (kurs.fach || "").trim();
  const name = (kurs.name || "").trim();
  if (!fach) return name;
  if (!name) return fach;
  // Steht das Fach schon im Namen („Mathe 7.5"), waere „Mathe · Mathe 7.5"
  // doppelt gemoppelt. Viele Konten benennen ihre Kurse genau so.
  if (name.toLowerCase().includes(fach.toLowerCase())) return name;
  return `${fach} · ${name}`;
}

File: apps/api/app/kurslabel.py
"""Wie ein Kurs beschriftet wird — ein Blatt, damit jeder Router es holen kann.

Stand bis 17.09.2026 als `_kurs_label` in routers/kalender.py. Das Notizbrett
braucht dieselbe Beschriftung (Korrektur-To-do „Mathe · 7.5: 2. KA
korrigieren"), darf den Kalender-Router aber nicht importieren: der importiert
schon `todos` — das waere der Importring. Kalender und CalDAV holen es weiter
unter dem alten Namen aus routers/kalender.py.

Gegenstueck: `kursLabel` in apps/web/src/core/kurslabel.js — zusammen aendern.
"""
from .kursmitglieder import kurs_der_klasse
from .models import Kurs


def kurs_label(kurs) -> str:
    """"Mathe · 7.5" — Fach zuerst, Kursname dahinter.

    Zwei Fassungen hiessen, dass derselbe Termin im Handykalender anders heisst
    als im Browser. Steht das Fach schon im Namen ("Mathe 7.5"), waere
    "Mathe · Mathe 7.5" doppelt gemoppelt — viele Konten benennen ihre Kurse
    genau so.
    """
    if kurs is None:
        return ""
    fach = (getattr(kurs, "fach", "") or "").strip()
    name = (getattr(kurs, "name", "") or "").strip()
    if not fach:
        return name
    if not name:
        return fach
    if fach.lower() in name.lower():
        return name
    return f"{fach} · {name}"


async def kurs_des_termins(db, owner_id: int, kurs_id, class_id):
    """Der Kurs eines Termins: sein eigener, sonst der EINE seiner Klasse.

    Mehrdeutig (Klasse in zwei Kursen) heisst None — geraten wird nicht.
    Fremde oder geloeschte Kurse zaehlen nicht.
    """
    kid = kurs_id or (await kurs_der_klasse(db, class_id) if class_id else None)
    if not kid:
        return None
    k = await db.get(Kurs, kid)
    if not k or k.owner_id != owner_id or getattr(k, "deleted_at", None) is not None:
        return None
    return k
